tt_metal/CMakeLists.txt

```diff
@@ -18,7 +18,7 @@ set(TT_METAL_OBJECTS
 
 add_library(tt_metal ${TT_METAL_OBJECTS})
 
-target_link_libraries(tt_metal PUBLIC umd_device metal_common_libs)
+target_link_libraries(tt_metal PUBLIC umd_device metal_common_libs reflect::reflect)
```

Comment:
Ayooooo chillll do we need to put reflect absolutely everywhere? Literally only profiler and tt_stl use it

Reply:
You see, metal depends on profiler, but not in a typical target_link_libraries way.
If that was it, reflect would propagate as a public dependency together with all required include paths.

@vtangTT , can you clarify here why we do it this way?

Reply:
@tt-rkim I just added it everywhere until it compiled lol. I am not sure how to clean this up

Reply:
To clean it up we switch things to use static/shared libraries instead of object libs and start to properly link them.
After this you don't have to manually propagate include dirs or libs.

@TT-billteng sounds good?

Reply:
Can you circumvent this by not linking the intf library and just target_include_directories the `${reflect_SOURCE_DIR}` that's provided by CPM in places that's needed. 

It might fix the problem of having to link the reflect intf library everywhere?

Reply:
@vtangTT can you please do it on my branch?

Reply:
@TT-billteng @vtangTT is it possible to merge this and clean it up after?

Reply:
man @arakhmati needs an intern to do his dirty work

Reply:
Should be clean now. @tt-rkim 

In the future, please try to use the intf libraries (`metal_common_libs`, `metal_header_directories`) provided at top level CMake like I did here 👍

Reply:
This looks better 👍